### ISSUE TITLE
Can't create a form using the "Contact Form" stencil

### ISSUE THREAD

**Issue by lindseydiloreto** [verbb/formie#2202]
### Describe the bug

When trying to create a new form using the "Contact Form" stencil, an error is thrown...

![Creating a new form](https://github.com/user-attachments/assets/a1a650f7-52b2-4f28-a5d6-f2cf63765986)

This was reported in the error log...

```
[ERROR] Couldn’t save form - [{"rows":[{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]},{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]},{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]}]},{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]},{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]},{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]},{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]},{"fields":[{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}]},{"label":["Label cannot be blank."],"handle":["Handle cannot be blank."]}].
```

Not sure what the issue is. I'm able to create "Blank" forms just fine.

### Steps to reproduce

1. Create a new form.
2. Select the "Contact Form" stencil.
3. Click "Next".

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

Craft Pro 5.5.7

### Plugin version

3.0.15

### Multi-site?

No

**Comment by lindseydiloreto** [comment]
When I edit the Stencil locally, it looks like this...

![Ω  2024-12-24  14 54 49](https://github.com/user-attachments/assets/455fd784-544a-498a-a4c1-eda0f5422a92)

**Comment by engram-design** [comment]
Strange, brand new install, or and upgrade? I've just tested with a brand new install, and that seems to be okay.

Maybe send through your stencil data, found in `formie_stencils` or in your project config `/config/project/formie/stencils` directory. Something is clearly a bit awry with your stencil.

Of course, updating those fields in the stencil should sort it out, but maybe a migration wasn't completed, so keen to hear what your situation is.

**Comment by lindseydiloreto** [comment]
This project has been through a lot, so it wouldn't surprise me if something weird happened along the way.

Here's the row in `formie_stencils`...

![row](https://github.com/user-attachments/assets/6425036a-384b-4aa6-ab2f-c2ffece09b0e)

And here's the `data` value specifically...

```js
{"dataRetention":"forever","dataRetentionValue":null,"fileUploadsAction":"retain","notifications":[{"attachAssets":null,"attachFiles":true,"attachPdf":null,"bcc":null,"cc":null,"conditions":null,"content":"[{\"type\":\"paragraph\",\"content\":[{\"type\":\"text\",\"text\":\"A user submission has been made on the \\\"\"},{\"type\":\"variableTag\",\"attrs\":{\"label\":\"Form Name\",\"value\":\"{formName}\"}},{\"type\":\"text\",\"text\":\"\\\" form on \"},{\"type\":\"variableTag\",\"attrs\":{\"label\":\"Site Name\",\"value\":\"{siteName}\"}},{\"type\":\"text\",\"text\":\" at \"},{\"type\":\"variableTag\",\"attrs\":{\"label\":\"Timestamp (yyyy-mm-dd hh:mm:ss)\",\"value\":\"{timestamp}\"}}]},{\"type\":\"paragraph\",\"content\":[{\"type\":\"text\",\"text\":\"Their submission details are:\"}]},{\"type\":\"paragraph\",\"content\":[{\"type\":\"variableTag\",\"attrs\":{\"label\":\"All Form Fields\",\"value\":\"{allFields}\"}}]}]","enableConditions":null,"enabled":true,"from":null,"fromName":null,"handle":"adminNotification","name":"Admin Notification","pdfTemplateId":null,"recipients":"email","replyTo":"{field:emailAddress}","replyToName":null,"sender":null,"subject":"A new submission was made on \"{formName}\"","templateId":null,"to":"{systemEmail}","toConditions":null},{"attachAssets":null,"attachFiles":true,"attachPdf":null,"bcc":null,"cc":null,"conditions":null,"content":"[{\"type\":\"paragraph\",\"content\":[{\"type\":\"text\",\"text\":\"Thanks again for contacting us. Our team will get back to you as soon as we can.\"}]},{\"type\":\"paragraph\",\"content\":[{\"type\":\"text\",\"text\":\"As a reminder, you submitted the following details at \"},{\"type\":\"variableTag\",\"attrs\":{\"label\":\"Timestamp (yyyy-mm-dd hh:mm:ss)\",\"value\":\"{timestamp}\"}}]},{\"type\":\"paragraph\",\"content\":[{\"type\":\"variableTag\",\"attrs\":{\"label\":\"All Form Fields\",\"value\":\"{allFields}\"}}]}]","enableConditions":null,"enabled":true,"from":null,"fromName":null,"handle":"userNotification","name":"User Notification","pdfTemplateId":null,"recipients":"email","replyTo":null,"replyToName":null,"sender":null,"subject":"Thanks for contacting us!","templateId":null,"to":"{field:emailAddress}","toConditions":null}],"pages":[{"label":"Page 1","rows":[{"fields":[{"settings":{"conditions":null,"containerAttributes":null,"cssClasses":null,"defaultValue":null,"emailValue":null,"enableConditions":false,"enableContentEncryption":false,"enabled":true,"errorMessage":"Component class 'verbb\\formie\\fields\\Name' belongs to an uninstalled plugin (Formie).","expectedType":"verbb\\formie\\fields\\Name","handle":null,"includeInEmail":true,"inputAttributes":null,"instructions":null,"instructionsPosition":null,"label":null,"labelPosition":null,"matchField":null,"placeholder":null,"prePopulate":null,"required":false,"settings":{"enabled":true,"handle":"yourName","instructions":"Please enter your full name.","instructionsPosition":"verbb\\formie\\positions\\AboveInput","label":"Your Name","labelPosition":"verbb\\formie\\positions\\Hidden","placeholder":"Your name","required":false,"rows":[{"fields":[{"settings":{"enabled":false,"handle":"prefix","inputAttributes":[{"label":"autocomplete","value":"honorific-prefix"}],"label":"Prefix","options":[{"isDefault":true,"isOptgroup":false,"label":"Select an option","value":""}],"required":false},"type":"verbb\\formie\\fields\\subfields\\NamePrefix"},{"settings":{"enabled":true,"handle":"firstName","inputAttributes":[{"label":"autocomplete","value":"given-name"}],"label":"First Name","placeholder":"e.g. Peter","required":true},"type":"verbb\\formie\\fields\\subfields\\NameFirst"},{"settings":{"enabled":false,"handle":"middleName","inputAttributes":[{"label":"autocomplete","value":"additional-name"}],"label":"Middle Name","required":false},"type":"verbb\\formie\\fields\\subfields\\NameMiddle"},{"settings":{"enabled":true,"handle":"lastName","inputAttributes":[{"label":"autocomplete","value":"family-name"}],"label":"Last Name","placeholder":"e.g. Sherman","required":true},"type":"verbb\\formie\\fields\\subfields\\NameLast"}]}],"useMultipleFields":true},"visibility":null},"type":"verbb\\formie\\fields\\MissingField"}]},{"fields":[{"settings":{"conditions":null,"containerAttributes":null,"cssClasses":null,"defaultValue":null,"emailValue":null,"enableConditions":false,"enableContentEncryption":false,"enabled":true,"errorMessage":"Component class 'verbb\\formie\\fields\\Email' belongs to an uninstalled plugin (Formie).","expectedType":"verbb\\formie\\fields\\Email","handle":null,"includeInEmail":true,"inputAttributes":null,"instructions":null,"instructionsPosition":null,"label":null,"labelPosition":null,"matchField":null,"placeholder":null,"prePopulate":null,"required":false,"settings":{"enabled":true,"handle":"emailAddress","instructions":"Please enter your email so we can get in touch.","label":"Email Address","placeholder":"e.g. psherman@wallaby.com","required":true},"visibility":null},"type":"verbb\\formie\\fields\\MissingField"}]},{"fields":[{"settings":{"conditions":null,"containerAttributes":null,"cssClasses":null,"defaultValue":null,"emailValue":null,"enableConditions":false,"enableContentEncryption":false,"enabled":true,"errorMessage":"Component class 'verbb\\formie\\fields\\MultiLineText' belongs to an uninstalled plugin (Formie).","expectedType":"verbb\\formie\\fields\\MultiLineText","handle":null,"includeInEmail":true,"inputAttributes":null,"instructions":null,"instructionsPosition":null,"label":null,"labelPosition":null,"matchField":null,"placeholder":null,"prePopulate":null,"required":false,"settings":{"enabled":true,"handle":"message","instructions":"Please enter your comments.","label":"Message","placeholder":"e.g. The reason for my enquiry is...","required":true},"visibility":null},"type":"verbb\\formie\\fields\\MissingField"}]}],"settings":{"backButtonLabel":"Back","buttonsPosition":"left","containerAttributes":null,"cssClasses":null,"enableJsEvents":false,"enableNextButtonConditions":false,"enablePageConditions":false,"inputAttributes":null,"saveButtonLabel":"Save","saveButtonStyle":"link","showBackButton":false,"showSaveButton":false,"submitButtonLabel":"Contact us"}}],"settings":{"collectIp":false,"collectUser":false,"dataRetention":null,"dataRetentionValue":null,"defaultEmailTemplateId":null,"defaultInstructionsPosition":"verbb\\formie\\positions\\AboveInput","defaultLabelPosition":"verbb\\formie\\positions\\AboveInput","disableCaptchas":false,"displayCurrentPageTitle":false,"displayFormTitle":false,"displayPageProgress":false,"displayPageTabs":false,"errorMessage":[{"content":[{"text":"Couldn’t save submission due to errors.","type":"text"}],"type":"paragraph"}],"errorMessagePosition":"top-form","fileUploadsAction":null,"limitSubmissions":false,"limitSubmissionsMessage":null,"limitSubmissionsNumber":null,"limitSubmissionsType":null,"loadingIndicator":null,"loadingIndicatorText":null,"pageRedirectUrl":null,"progressPosition":"end","redirectUrl":null,"requireUser":false,"requireUserMessage":null,"requiredIndicator":"asterisk","scheduleForm":false,"scheduleFormEnd":null,"scheduleFormExpiredMessage":null,"scheduleFormPendingMessage":null,"scheduleFormStart":null,"scrollToTop":true,"submissionTitleFormat":"{timestamp}","submitAction":null,"submitActionFormHide":false,"submitActionMessage":[{"content":[{"text":"Submission saved.","type":"text"}],"type":"paragraph"}],"submitActionMessagePosition":"top-form","submitActionMessageTimeout":null,"submitActionTab":null,"submitActionUrl":null,"submitMethod":null,"validationOnFocus":false,"validationOnSubmit":true},"userDeletedAction":"retain"}
```

And here's what I found in the project config file...

```yaml
data:
  dataRetention: forever
  dataRetentionValue: null
  fileUploadsAction: retain
  notifications:
    -
      attachAssets: null
      attachFiles: true
      attachPdf: null
      bcc: null
      cc: null
      conditions: null
      content: '[{"type":"paragraph","content":[{"type":"text","text":"A user submission has been made on the \""},{"type":"variableTag","attrs":{"label":"Form Name","value":"{formName}"}},{"type":"text","text":"\" form on "},{"type":"variableTag","attrs":{"label":"Site Name","value":"{siteName}"}},{"type":"text","text":" at "},{"type":"variableTag","attrs":{"label":"Timestamp (yyyy-mm-dd hh:mm:ss)","value":"{timestamp}"}}]},{"type":"paragraph","content":[{"type":"text","text":"Their submission details are:"}]},{"type":"paragraph","content":[{"type":"variableTag","attrs":{"label":"All Form Fields","value":"{allFields}"}}]}]'
      enableConditions: null
      enabled: true
      from: null
      fromName: null
      handle: adminNotification
      name: 'Admin Notification'
      pdfTemplateId: null
      recipients: email
      replyTo: '{field:emailAddress}'
      replyToName: null
      sender: null
      subject: 'A new submission was made on "{formName}"'
      templateId: null
      to: '{systemEmail}'
      toConditions: null
    -
      attachAssets: null
      attachFiles: true
      attachPdf: null
      bcc: null
      cc: null
      conditions: null
      content: '[{"type":"paragraph","content":[{"type":"text","text":"Thanks again for contacting us. Our team will get back to you as soon as we can."}]},{"type":"paragraph","content":[{"type":"text","text":"As a reminder, you submitted the following details at "},{"type":"variableTag","attrs":{"label":"Timestamp (yyyy-mm-dd hh:mm:ss)","value":"{timestamp}"}}]},{"type":"paragraph","content":[{"type":"variableTag","attrs":{"label":"All Form Fields","value":"{allFields}"}}]}]'
      enableConditions: null
      enabled: true
      from: null
      fromName: null
      handle: userNotification
      name: 'User Notification'
      pdfTemplateId: null
      recipients: email
      replyTo: null
      replyToName: null
      sender: null
      subject: 'Thanks for contacting us!'
      templateId: null
      to: '{field:emailAddress}'
      toConditions: null
  pages:
    -
      label: 'Page 1'
      rows:
        -
          fields:
            -
              settings:
                conditions: null
                containerAttributes: null
                cssClasses: null
                defaultValue: null
                emailValue: null
                enableConditions: false
                enableContentEncryption: false
                enabled: true
                errorMessage: 'Component class ''verbb\formie\fields\Name'' belongs to an uninstalled plugin (Formie).'
                expectedType: verbb\formie\fields\Name
                handle: null
                includeInEmail: true
                inputAttributes: null
                instructions: null
                instructionsPosition: null
                label: null
                labelPosition: null
                matchField: null
                placeholder: null
                prePopulate: null
                required: false
                settings:
                  enabled: true
                  handle: yourName
                  instructions: 'Please enter your full name.'
                  instructionsPosition: verbb\formie\positions\AboveInput
                  label: 'Your Name'
                  labelPosition: verbb\formie\positions\Hidden
                  placeholder: 'Your name'
                  required: false
                  rows:
                    -
                      fields:
                        -
                          settings:
                            enabled: false
                            handle: prefix
                            inputAttributes:
                              -
                                label: autocomplete
                                value: honorific-prefix
                            label: Prefix
                            options:
                              -
                                isDefault: true
                                isOptgroup: false
                                label: 'Select an option'
                                value: ''
                            required: false
                          type: verbb\formie\fields\subfields\NamePrefix
                        -
                          settings:
                            enabled: true
                            handle: firstName
                            inputAttributes:
                              -
                                label: autocomplete
                                value: given-name
                            label: 'First Name'
                            placeholder: 'e.g. Peter'
                            required: true
                          type: verbb\formie\fields\subfields\NameFirst
                        -
                          settings:
                            enabled: false
                            handle: middleName
                            inputAttributes:
                              -
                                label: autocomplete
                                value: additional-name
                            label: 'Middle Name'
                            required: false
                          type: verbb\formie\fields\subfields\NameMiddle
                        -
                          settings:
                            enabled: true
                            handle: lastName
                            inputAttributes:
                              -
                                label: autocomplete
                                value: family-name
                            label: 'Last Name'
                            placeholder: 'e.g. Sherman'
                            required: true
                          type: verbb\formie\fields\subfields\NameLast
                  useMultipleFields: true
                visibility: null
              type: verbb\formie\fields\MissingField
        -
          fields:
            -
              settings:
                conditions: null
                containerAttributes: null
                cssClasses: null
                defaultValue: null
                emailValue: null
                enableConditions: false
                enableContentEncryption: false
                enabled: true
                errorMessage: 'Component class ''verbb\formie\fields\Email'' belongs to an uninstalled plugin (Formie).'
                expectedType: verbb\formie\fields\Email
                handle: null
                includeInEmail: true
                inputAttributes: null
                instructions: null
                instructionsPosition: null
                label: null
                labelPosition: null
                matchField: null
                placeholder: null
                prePopulate: null
                required: false
                settings:
                  enabled: true
                  handle: emailAddress
                  instructions: 'Please enter your email so we can get in touch.'
                  label: 'Email Address'
                  placeholder: 'e.g. psherman@wallaby.com'
                  required: true
                visibility: null
              type: verbb\formie\fields\MissingField
        -
          fields:
            -
              settings:
                conditions: null
                containerAttributes: null
                cssClasses: null
                defaultValue: null
                emailValue: null
                enableConditions: false
                enableContentEncryption: false
                enabled: true
                errorMessage: 'Component class ''verbb\formie\fields\MultiLineText'' belongs to an uninstalled plugin (Formie).'
                expectedType: verbb\formie\fields\MultiLineText
                handle: null
                includeInEmail: true
                inputAttributes: null
                instructions: null
                instructionsPosition: null
                label: null
                labelPosition: null
                matchField: null
                placeholder: null
                prePopulate: null
                required: false
                settings:
                  enabled: true
                  handle: message
                  instructions: 'Please enter your comments.'
                  label: Message
                  placeholder: 'e.g. The reason for my enquiry is...'
                  required: true
                visibility: null
              type: verbb\formie\fields\MissingField
      settings:
        backButtonLabel: Back
        buttonsPosition: left
        containerAttributes: null
        cssClasses: null
        enableJsEvents: false
        enableNextButtonConditions: false
        enablePageConditions: false
        inputAttributes: null
        saveButtonLabel: Save
        saveButtonStyle: link
        showBackButton: false
        showSaveButton: false
        submitButtonLabel: 'Contact us'
  settings:
    collectIp: false
    collectUser: false
    dataRetention: null
    dataRetentionValue: null
    defaultEmailTemplateId: null
    defaultInstructionsPosition: verbb\formie\positions\AboveInput
    defaultLabelPosition: verbb\formie\positions\AboveInput
    disableCaptchas: false
    displayCurrentPageTitle: false
    displayFormTitle: false
    displayPageProgress: false
    displayPageTabs: false
    errorMessage:
      -
        content:
          -
            text: 'Couldn’t save submission due to errors.'
            type: text
        type: paragraph
    errorMessagePosition: top-form
    fileUploadsAction: null
    limitSubmissions: false
    limitSubmissionsMessage: null
    limitSubmissionsNumber: null
    limitSubmissionsType: null
    loadingIndicator: null
    loadingIndicatorText: null
    pageRedirectUrl: null
    progressPosition: end
    redirectUrl: null
    requireUser: false
    requireUserMessage: null
    requiredIndicator: asterisk
    scheduleForm: false
    scheduleFormEnd: null
    scheduleFormExpiredMessage: null
    scheduleFormPendingMessage: null
    scheduleFormStart: null
    scrollToTop: true
    submissionTitleFormat: '{timestamp}'
    submitAction: null
    submitActionFormHide: false
    submitActionMessage:
      -
        content:
          -
            text: 'Submission saved.'
            type: text
        type: paragraph
    submitActionMessagePosition: top-form
    submitActionMessageTimeout: null
    submitActionTab: null
    submitActionUrl: null
    submitMethod: null
    validationOnFocus: false
    validationOnSubmit: true
  userDeletedAction: retain
defaultStatus: 787dc37a-2f2a-4b7c-8d15-626170731878 # New
handle: contactForm
name: 'Contact Form'
submitActionEntry: null
template: null
```

**Comment by lindseydiloreto** [comment]
No rush, btw... I just started with a blank form and built what I needed.

Thanks for such a quick response over a holiday! Merry Christmas @engram-design! 🎄🎅

**Comment by engram-design** [comment]
Looks like the missing field information has been incorrectly stored into a stencils data. There should be handling for converting it back from a missing field. At some point, somehow - the stencil has determined these Formie fields are missing.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.15"`.

**Comment by engram-design** [comment]
Fixed in [3.0.16](https://github.com/verbb/formie/releases/tag/3.0.16)

**Comment by lindseydiloreto** [comment]
Brilliant, thanks @engram-design! 🍺
